packages/core/src/effects/effects.interface.ts

```diff
@@ -13,8 +16,12 @@ export interface Middleware<
 > extends Effect<I, O> {}
 
 export interface ErrorEffect<T extends Error = Error>
-  extends Effect<HttpRequest, EffectResponse, T> {}
+  extends Effect<HttpRequest, EffectHttpResponse, T> {}
```

Comment:
Since it extends Effect<HttpRequest, EffectHttpResponse>, shouldn't it be HttpErrorEffect?

Reply:
If I will name it as `HttpErrorEffect` then it will be a huge breaking change because constructs like `Effect`, `ErrorEffect`, `Middleware` are more commonly used compared to raw `Error<..., ...> definitions. 🤔

Reply:
Maybe `ErrorEffect` should be more generic?

Reply:
Or to only define the fully compassable `Effect` interfaces and leave `ErrorEffect` and `Middleware` as it is. 

```typescript
export interface Effect<
  T = HttpRequest,
  U = EffectHttpResponse,
  V = any,
  W = HttpResponse,
> {
  (req$: Observable<T>, res: W, meta: V): Observable<U>;
}
```

Then we can create `WsEffect`, `WsErrorEffect` and `WsMiddleware` types basing only on ☝️ it.

Reply:
IMO the better is the last solution. 🤔

Reply:
Hm, I think that the latter solution would be best. Default way of working with Marble would be HTTP, and WS is added as additional feature. 👌 BTW. Should we add support for server push (with http2) in Marble.js 2.0?

Reply:
Yes, but these are our priorities for v2.0:
`Websockets` >> `http2`